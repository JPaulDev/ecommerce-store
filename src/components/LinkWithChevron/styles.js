import styled from 'styled-components';

export const Link = styled.a`
  align-items: center;
  color: var(--color-grey-8);
  display: flex;
  font-size: ${({ $fontSize }) => $fontSize};
  font-weight: 700;
  text-decoration: none;
  text-transform: uppercase;
`;

export const Container = styled.div`
  background-color: var(--color-grey-8);
  border-radius: 50%;
  display: flex;
  margin-left: 10px;
`;
