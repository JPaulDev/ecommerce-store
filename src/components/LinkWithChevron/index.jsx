import MaterialIcon from '../MaterialIcon';
import * as Styled from './styles';

export default function LinkWithChevron({ url, text, fontSize, iconSize }) {
  return (
    <Styled.Link href={url} $fontSize={fontSize} data-testid="link-chevron">
      {text}
      <MaterialIcon iconSize={iconSize} iconColor="white">
        chevron_right
      </MaterialIcon>
    </Styled.Link>
  );
}
