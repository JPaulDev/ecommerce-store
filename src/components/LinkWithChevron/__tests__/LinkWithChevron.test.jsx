import renderer from 'react-test-renderer';
import LinkWithChevron from '../index';

describe('LinkWithChevron component', () => {
  it('renders a link with a chevron icon', () => {
    const tree = renderer
      .create(<LinkWithChevron url="test url" text="test text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
