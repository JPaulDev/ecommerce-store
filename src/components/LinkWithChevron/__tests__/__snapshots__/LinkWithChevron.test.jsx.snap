// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkWithChevron component renders a link with a chevron icon 1`] = `
<a
  className="sc-gsnTZi FwZzn"
  data-testid="link-chevron"
  href="test url"
>
  test text
  <div
    className="sc-dkzDqf iNOEfq"
  >
    <i
      className="sc-bczRLJ klUGFp material-icons"
    >
      chevron_right
    </i>
  </div>
</a>
`;
